improver_tests/nbhood/nbhood/test_GeneratePercentilesFromANeighbourhood.py

```diff
@@ -387,18 +386,38 @@ def test_single_point(self):
         ).process(cube)
         self.assertArrayAlmostEqual(result.data, expected)
 
-    def test_multi_point_multitimes(self):
-        """Test behaviour for points over multiple times."""
+    def test_single_point_single_percentile(self):
+        """Test behaviour for a single non-zero grid cell."""
+
+        expected = np.array(
+            [
+                [
+                    [1.0, 1.0, 1.0, 1.0, 1.0],
+                    [1.0, 1.0, 1.0, 1.0, 1.0],
+                    [1.0, 1.0, 1.0, 1.0, 1.0],
+                    [1.0, 1.0, 1.0, 1.0, 1.0],
+                    [1.0, 1.0, 1.0, 1.0, 1.0],
+                ]
+            ]
+        )
 
         data = np.ones((5, 5), dtype=np.float32)
+        data[2, 2] = 0
         cube = set_up_variable_cube(data, spatial_grid="equalarea",)
-        time_points = [
-            datetime(2017, 11, 10, 2),
-            datetime(2017, 11, 10, 3),
-        ]
-        cube = add_coordinate(
-            cube, coord_points=time_points, coord_name="time", is_datetime="true",
-        )
+        percentiles = np.array([50])
+        radius = 2000.0
+        result = GeneratePercentilesFromANeighbourhood(
+            radius, percentiles=percentiles
+        ).process(cube)
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_multi_point_multireals(self):
+        """Test behaviour for points over multiple realizations."""
+
+        data = np.ones((5, 5), dtype=np.float32)
+        cube = set_up_variable_cube(data, spatial_grid="equalarea",)
+        reals_points = np.array([0, 1], dtype=np.int32)
```

Comment:
I've modified this test to improve test coverage. We don't support multitime coordinates in improver, but you could have multiple realizations. This test now uses an input cube with two realizations, and creates an output cube with a leading dimension of realization followed by the percentile dimension.